src/Microsoft.AspNetCore.Testing/xunit/MinOsVersionSkipConditionAttribute.cs

```diff
@@ -0,0 +1,109 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Linq;
+using System.Runtime.InteropServices;
+using Microsoft.Win32;
+
+namespace Microsoft.AspNetCore.Testing.xunit
+{
+    /// <summary>
+    /// Skips a test if the OS is the given type (Windows) and the OS version is less than specified.
+    /// E.g. Specifying Window 10.0 skips on Win 8, but not on Linux. Combine with OSSkipConditionAttribute as needed.
+    /// </summary>
+    [AttributeUsage(AttributeTargets.Method | AttributeTargets.Class | AttributeTargets.Assembly, AllowMultiple = true)]
+    public class MinOsVersionSkipConditionAttribute : Attribute, ITestCondition
+    {
+        private readonly OperatingSystems _excludedOperatingSystem;
+        private readonly Version _minVersion;
+        private readonly OperatingSystems _osPlatform;
+        private readonly Version _osVersion;
+
+        public MinOsVersionSkipConditionAttribute(OperatingSystems operatingSystem, string minVersion) :
+            this(
+                operatingSystem,
+                GetCurrentOS(),
+                GetCurrentOSVersion(),
+                Version.Parse(minVersion))
+        {
+        }
+
+        // to enable unit testing
+        internal MinOsVersionSkipConditionAttribute(
+            OperatingSystems operatingSystem, OperatingSystems osPlatform, Version osVersion, Version minVersion)
+        {
+            if (operatingSystem != OperatingSystems.Windows)
+            {
+                throw new NotImplementedException("Min version support is only implemented for Windows.");
+            }
+            _excludedOperatingSystem = operatingSystem;
+            _minVersion = minVersion;
+            _osPlatform = osPlatform;
+            _osVersion = osVersion;
+        }
+
+        public bool IsMet
+        {
+            get
+            {
+                // Do not skip other OS's, Use OSSkipConditionAttribute or a separate MinOsVersionSkipConditionAttribute for that.
+                if (_osPlatform != _excludedOperatingSystem)
+                {
+                    return true;
+                }
+
+                return _osVersion >= _minVersion;
+            }
+        }
+
+        public string SkipReason { get; set; } = "Test cannot run on this operating system.";
```

Comment:
Could we take the opportunity to make this a little more expressive?

`This test requires [OSPlatform] [Version] or higher`